src/main/java/frc/robot/commands/autonomousCommands/AutonomousShooterCommand.java

```diff
@@ -25,6 +26,9 @@
     private IndexerController indexerController;
     //variables
     private double ballsShot = 0;
+    private boolean ballShot = false;
+    private double startTime;
+    private boolean ballDetected = false;
```

Comment:
you don't need a class member for ballDetected.  Instead you'll want a local DistanceSensorDetector, like the IndexerController.
Make sure the constructor takes it as an input, and you'll have to add it to constructors up the chain.
Then whenever you want to know if there is a ball detected, just call distanceSensorDetector.isBallDetected()